## Description: Fix build in amd64, due to casting gpointers to (gint). Now we use the GPOINTER_TO_INT macro wherever it's needed.
## Origin: upstream, no
## Author: Pojar George <geoubuntu@gmail.com> 
## Forwarded: no
Index: gnomeradio-1.8/src/gui.c
===================================================================
--- gnomeradio-1.8.orig/src/gui.c
+++ gnomeradio-1.8/src/gui.c
@@ -537,7 +537,7 @@ static gboolean poll_volume_change(gpoin
 
 static void change_frequency(gpointer data)
 {
-	gboolean increase = (gboolean)data;
+	gboolean increase = (gboolean) GPOINTER_TO_INT (data);
 	gint v = gtk_adjustment_get_value(adj);
 	
 	if (increase)
@@ -590,7 +590,7 @@ static void step_button_released_cb(GtkB
 static gboolean scan_freq(gpointer data)
 {
 	static gint start, mom, max;
-	gint dir = (gint)(data);
+	gint dir = GPOINTER_TO_INT (data);
 	
 	if (!max) {
 		max = (FREQ_MAX - FREQ_MIN) * STEPS;
Index: gnomeradio-1.8/src/prefs.c
===================================================================
--- gnomeradio-1.8.orig/src/prefs.c
+++ gnomeradio-1.8/src/prefs.c
@@ -322,7 +322,7 @@ static void add_button_clicked_cb(GtkWid
 		menuitem = gtk_menu_item_new_with_label(ps->title); 
 			
 		gtk_menu_shell_insert(GTK_MENU_SHELL(tray_menu), menuitem, mom_ps);		
-		g_signal_connect(G_OBJECT(menuitem), "activate", (GCallback)preset_menuitem_activate_cb, (gpointer)mom_ps);
+		g_signal_connect(G_OBJECT(menuitem), "activate", (GCallback)preset_menuitem_activate_cb, GINT_TO_POINTER (mom_ps));
 		gtk_widget_show(menuitem);
 	}
 
@@ -430,7 +430,7 @@ static void name_cell_edited_cb(GtkCellR
 		menuitem = gtk_menu_item_new_with_label(ps->title); 
 			
 		gtk_menu_shell_insert(GTK_MENU_SHELL(tray_menu), menuitem, *row);		
-		g_signal_connect(G_OBJECT(menuitem), "activate", (GCallback)preset_menuitem_activate_cb, (gpointer)mom_ps);
+		g_signal_connect(G_OBJECT(menuitem), "activate", (GCallback)preset_menuitem_activate_cb, GINT_TO_POINTER (mom_ps));
 		gtk_widget_show(menuitem);
 	}
 	
Index: gnomeradio-1.8/src/trayicon.c
===================================================================
--- gnomeradio-1.8.orig/src/trayicon.c
+++ gnomeradio-1.8/src/trayicon.c
@@ -56,7 +56,7 @@ static void quit_menuitem_activate_cb(Gt
 void preset_menuitem_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
 {
 	preset* ps;
-	mom_ps = (int)user_data;
+	mom_ps = GPOINTER_TO_INT (user_data);
 	
 	g_assert(mom_ps >= 0 &&	mom_ps < g_list_length(settings.presets));
 	
@@ -76,7 +76,7 @@ void create_tray_menu(GtkWidget *app) {
 		GtkWidget *menuitem = gtk_menu_item_new_with_label(ps->title); 
 		
 		gtk_menu_shell_insert(GTK_MENU_SHELL(tray_menu), menuitem, i);		
-		g_signal_connect(G_OBJECT(menuitem), "activate", (GCallback)preset_menuitem_activate_cb, (gpointer)i);
+		g_signal_connect(G_OBJECT(menuitem), "activate", (GCallback)preset_menuitem_activate_cb, GINT_TO_POINTER (i));
 		gtk_widget_show(menuitem);
 	}
 	
